### ISSUE TITLE
Disabling BLE client as Cannot find package '@abandonware/bluetooth-hci-socket' imported from /var/lib/homebridge/node_modules/homebridge-govee/lib/platform.js at new NodeError (node:internal/errors:387:5).

### ISSUE THREAD

**Issue by redvegas** [bwp91/homebridge-govee#331]
### What issue do you have? Please be as thorough and explicit as possible.
Hello,

I am trying to connect govee led light strip H617A.

they show up in the log but when I try to add the device ID to the plugin nothing comes up. I followed all the instructions and tried finding the BLE on my own but noticed I was getting the following error and was not able to debug the lights and that's most likely the reason the light aren't connecting in the first place.
[10/16/2022, 1:57:20 PM] [Govee] [Bed light] [27:6D:D7:37:32:34:73:61] will not be added to Homebridge unless configured and Bluetooth enabled in config.
[10/16/2022, 1:57:20 PM] [Govee] [TV Light] [04:FE:D7:37:32:34:3C:88] will not be added to Homebridge unless configured and Bluetooth enabled in config.

### Details of your setup.
* Do you use (1) Homebridge UI-X (2) Homebridge CLI or (3) HOOBS? 
Homebridge

* Which version of Homebridge/HOOBS do you have?
[v1.5.0]

* Which platform do you run Homebridge/HOOBS on (e.g. Raspberry Pi/Windows/HOOBS Box)? Please also mention your version of Node.js/NPM if known.

[Banana Pi BPI-M2 Zero - Quad Core Single Board Computer with Wifi & Bluetooth]
 on Raspbian 9 OS.

* Which version of this plugin (homebridge-govee) do you have? Has the issue started since upgrading from a previous version?

homebridge-govee v6.0.4

* Which Govee devices do you have that are causing issues? Please include product models if applicable.

govee led light strip H617A.

### Please paste any relevant logs below.

```
10/16/2022, 1:57:13 PM] [Govee] Initializing Govee platform...
[10/16/2022, 1:57:13 PM] [Govee] Initialising plugin v6.0.4 | System linux | Node v16.18.0 | HB v1.5.0 | HAPNodeJS v0.10.4...
[10/16/2022, 1:57:13 PM] Loading 2 accessories...
[10/16/2022, 1:57:13 PM] [Unlock Front Door] Initializing DummySwitch accessory...
[10/16/2022, 1:57:14 PM] [Lock Front Door] Initializing DelaySwitch accessory...
[10/16/2022, 1:57:14 PM] [Govee] Plugin initialised. Setting up accessories....
Setup Payload:
X-HM://00250E7LDZPY5
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 931-55-825 │     
    └────────────┘     
                       
[10/16/2022, 1:57:14 PM] Homebridge v1.5.0 (HAP v0.10.4) (Homebridge 8A48) is running on port 51781.
[10/16/2022, 1:57:16 PM] [Govee] LAN client enabled.
[10/16/2022, 1:57:18 PM] [Govee] HTTP/AWS clients enabled as Govee username and password provided.
[10/16/2022, 1:57:19 PM] [Govee] API client enabled as Govee API key provided.
[10/16/2022, 1:57:19 PM] [Govee] Disabling BLE client as Cannot find package '@abandonware/bluetooth-hci-socket' imported from /var/lib/homebridge/node_modules/homebridge-govee/lib/platform.js at new NodeError (node:internal/errors:387:5).
[10/16/2022, 1:57:20 PM] [Govee] [Bed light] [27:6D:D7:37:32:34:73:61] will not be added to Homebridge unless configured and bluetooth enabled in config.
[10/16/2022, 1:57:20 PM] [Govee] [TV Light] [04:FE:D7:37:32:34:3C:88] will not be added to Homebridge unless configured and bluetooth enabled in config.
[10/16/2022, 1:57:20 PM] [Govee] [RGBICW Floor Lamp Basic] if you notice any flickering when changing the brightness/colour, let me know on Discord/Github with model number [H6076].
[10/16/2022, 1:57:20 PM] [Govee] [RGBICW Floor Lamp Basic] initialising with options {"adaptiveLightingShift":0,"aws":"enabled","ble":"disabled","brightnessStep":1,"lan":"enabled","logging":"standard"}.
[10/16/2022, 1:57:20 PM] [Govee] [RGBICW Floor Lamp Basic] initialised with id [36:7E:C1:37:34:32:38:19] [H6076].
[10/16/2022, 1:57:22 PM] [Govee] ✓ Setup complete. Want to see this plugin in your own language? Let me know
```

**Comment by bwp91** [comment]
Hi @redvegas 

Looks like a dependency that the plugin needs for bluetooth is failing to install.

Can you run this command in the homebridge terminal and paste the output? it may well be quite long!

npm install @abandonware/bluetooth-hci-socket

**Comment by redvegas** [comment]
Here is the output.

npm WARN checkPermissions Missing write access to /home/pi/node_modules
npm WARN enoent ENOENT: no such file or directory, open '/home/pi/package.json'
npm WARN node-fetch@2.6.7 requires a peer of encoding@^0.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN pi No description
npm WARN pi No repository field.
npm WARN pi No README data
npm WARN pi No license field.

npm ERR! path /home/pi/node_modules
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall access
npm ERR! Error: EACCES: permission denied, access '/home/pi/node_modules'
npm ERR!  { Error: EACCES: permission denied, access '/home/pi/node_modules'
npm ERR!   stack: 'Error: EACCES: permission denied, access \'/home/pi/node_modules\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/home/pi/node_modules' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2022-10-16T11_57_00_783Z-debug.log

**Comment by redvegas** [comment]
npm WARN checkPermissions Missing write access to /home/pi/node_modules/agent-base/node_modules/ms
npm WARN checkPermissions Missing write access to /home/pi/node_modules/agent-base/node_modules/debug
npm WARN checkPermissions Missing write access to /home/pi/node_modules/https-proxy-agent/node_modules/ms
npm WARN checkPermissions Missing write access to /home/pi/node_modules/https-proxy-agent/node_modules/debug
npm WARN checkPermissions Missing write access to /home/pi/node_modules/@abandonware/bluetooth-hci-socket/node_modules
npm WARN checkPermissions Missing write access to /home/pi/node_modules/agent-base/node_modules
npm WARN checkPermissions Missing write access to /home/pi/node_modules/https-proxy-agent/node_modules
npm WARN checkPermissions Missing write access to /home/pi/node_modules/@abandonware
npm WARN checkPermissions Missing write access to /home/pi/node_modules
npm WARN checkPermissions Missing write access to /home/pi/node_modules/bluetooth-hci-socket
npm WARN checkPermissions Missing write access to /home/pi/node_modules/@mapbox
npm WARN checkPermissions Missing write access to /home/pi/node_modules/make-dir/node_modules
npm WARN enoent ENOENT: no such file or directory, open '/home/pi/package.json'
npm WARN node-fetch@2.6.7 requires a peer of encoding@^0.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN pi No description
npm WARN pi No repository field.
npm WARN pi No README data
npm WARN pi No license field.

npm ERR! path /home/pi/node_modules/agent-base/node_modules/ms
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall access
npm ERR! Error: EACCES: permission denied, access '/home/pi/node_modules/agent-base/node_modules/ms'
npm ERR!  { Error: EACCES: permission denied, access '/home/pi/node_modules/agent-base/node_modules/ms'
npm ERR!   stack: 'Error: EACCES: permission denied, access \'/home/pi/node_modules/agent-base/node_modules/ms\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/home/pi/node_modules/agent-base/node_modules/ms' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2022-10-16T12_02_34_622Z-debug.log

**Comment by bwp91** [comment]
Can you next show the output of running this?

sudo npm install @abandonware/bluetooth-hci-socket

**Comment by cros-nash** [comment]
Hi there,

I had the same issue and after looking through various forums, I managed to solve it.

I ran these commands and it worked:

cd /var/lib/homebridge/node_modules/homebridge-govee/node_modules/

npm install @abandonware/bluetooth-hci-socket

**Comment by redvegas** [comment]
> Can you next show the output of running this?
> 
> sudo npm install @abandonware/bluetooth-hci-socket

npm install @abandonware/bluetooth-hci-socket
node: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.27' not found (required by node)
node: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.28' not found (required by node)
node: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.25' not found (required by node)

**Comment by redvegas** [comment]
> npm install @abandonware/bluetooth-hci-socket

/var/lib/homebridge/node_modules/homebridge-govee/node_modules $ npm install @abandonware/bluetooth-hci-socket
node: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.27' not found (required by node)
node: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.28' not found (required by node)
node: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.25' not found (required by node)

**Comment by redvegas** [comment]
Hello All,

I have decided to switch from using Raspbian 2020-04-10 update Raspbian Stretch, kernel 4.4 to Armbian_22.11.0-trunk_Bananapim2zero_jammy_edge_5.19.6_xfce_desktop. 
After installing HomeBridge, I did have to use the following commands to get the BLE to work but after that everything was working with no errors in HomeBridge logs. Thanks to @cros-nash for providing this.

cd /var/lib/homebridge/node_modules/homebridge-govee/node_modules/

npm install @abandonware/bluetooth-hci-socket

After adding all my Govee lights with the HomeBridge Govee plugin and setting up the Bluetoooth IDs under the lights setting in the Govee plugin I was able to successfully see all my lights automatically added in the accessory section. 

Unfortunately, after all of this I still am having an issue. 
My Banana Pi BPI-M2 ZERO doesn't seem to have enough Bluetooth signal strength to control the Govee_H617A_7361 lights. I try to turn the lights on and off from the accessory tab in HomeBridge and the lights will respond 1/15 try's clicking on/off.
This is what I am seeing in logs as well.

[TV Light] could not find device [d7:37:32:34:3c:88].
[10/30/2022, 10:26:14 PM] [Govee] [TV Light] could not be updated as no connection method available at file:///var/lib/homebridge/node_modules/homebridge-govee/lib/platform.js:1752:13.
[Bed light] could not find device [d7:37:32:34:73:61].
[10/30/2022, 10:27:31 PM] [Govee] [Bed light] could not be updated as no connection method available at file:///var/lib/homebridge/node_modules/homebridge-govee/lib/platform.js:1752:13.

I would like to confirm I do have a Govee_H6076_3819 that uses WiFi to control, and this light responds instantly. this issue only seems to be with Bluetooth.

I have purchased a micro-USB to USB adapter and Bluetooth adapter that I am going to try today and hopefully that will improve the Bluetooth signal and control the lights much better I will keep my post updated and inform if this gets me from 80% to 100% functionality.
